packages/ledger/src/index.ts

```diff
@@ -0,0 +1,270 @@
+  const { publicToAddress, toChecksumAddress } = await import('ethereumjs-util')
+
+  const node: BIP32Interface = BIP32Factory(ecc).fromPublicKey(
+    Buffer.from(publicKey, 'hex'),
+    Buffer.from(chainCode, 'hex')
+  )
+
+  const child: BIP32Interface = node.derive(index)
+
+  const address = toChecksumAddress(
+    `0x${publicToAddress(child.publicKey, true).toString('hex')}`
+  )
+
+  console.log({
+    address,
+    index,
+    derivationPath,
+    fullPath: `${derivationPath}/${index}`
+  })
+
+  return {
+    derivationPath,
+    address,
+    balance: {
+      asset: asset.label,
+      value: await getBalance(address, rpcUrl)
+    }
+  }
+}
+
+const getAddresses = (
+  account: LedgerAccount,
+  asset: Asset,
+  currentChain: Chain,
+  offset: number = 0,
+  limit: number = 15
+): Promise<Account[]> =>
+  limit - offset <= 0
+    ? (() => {
+        throw new Error('Offset must be less than limit')
+      })()
+    : Promise.all(
+        Array.from({ length: limit - offset }, (_, index) =>
+          getAddress(account, asset, currentChain, index)
+        )
+      )
+
+function ledger(): WalletInit {
+  return () => {
+    let accounts: Account[] | undefined
+    return {
+      label: 'Ledger',
+      getIcon: async () => (await import('./icon')).default,
+      getInterface: async ({ EventEmitter, chains }) => {
+        const Eth = (await import('@ledgerhq/hw-app-eth')).default
+        const { TransactionFactory: Transaction, Capability } = await import(
+          '@ethereumjs/tx'
+        )
+        const { default: Common, Hardfork } = await import('@ethereumjs/common')
+        const { rlp } = (await import('ethereumjs-util')).default
+
+        const transport: Transport = await getTransport()
+        const eth = new Eth(transport)
+        const eventEmitter = new EventEmitter()
+
+        let currentChain: Chain | undefined
+        const scanAccounts = async ({
+          derivationPath,
+          chainId,
+          asset
+        }: ScanAccountsOptions): Promise<Account[]> => {
+          currentChain = chains.find(({ id }) => id === chainId)
+          // TODO: Maybe throw error here
+          if (!currentChain) return []
+
+          const { publicKey, chainCode } = await eth.getAddress(
+            derivationPath,
+            false,
+            true // set to true to return chainCode
+          )
+
+          const { compress } = (await import('eth-crypto')).publicKey
+
+          return getAddresses(
+            {
+              publicKey: compress(publicKey),
+              chainCode: chainCode ?? '',
+              derivationPath: derivationPath
+            },
+            asset,
+            currentChain
+          )
+        }
+
+        const getAccounts = async () =>
+          accounts ??
+          (accounts = await accountSelect({
+            basePaths: DEFAULT_BASE_PATHS,
+            assets,
+            chains,
+            scanAccounts,
+            walletIcon: '<svg></svg>'
```

Comment:
@aaronbarnardsound Wasn't sure how this is different from the `getIcon` method above? 🤔

Reply:
Yeah essentially the accountSelect module needs the wallet icon to render it in the modal. If you could call the `getIcon` method and then pass the icon in here that would be great.